pkg/acsengine/engine.go

```diff
@@ -295,15 +300,27 @@ func prepareTemplateFiles(properties *api.Properties) ([]string, string, error)
 //for example: if the target is the public azure, then the default container image url should be gcr.io/google_container/...
 //if the target is azure china, then the default container image should be mirror.azure.cn:5000/google_container/...
 func GetCloudSpecConfig(location string) AzureEnvironmentSpecConfig {
-	cloudSpecConfig := AzureCloudSpec
-	switch location {
-	case "AzureCloud":
-		cloudSpecConfig = AzureCloudSpec
-	case "AzureChinaCloud":
-		cloudSpecConfig = AzureChinaCloudSpec
+	switch GetCloudTargetEnv(location) {
+	case azureCloudChina:
+		return AzureChinaCloudSpec
+	default:
+		return AzureCloudSpec
 	}
+}
 
-	return cloudSpecConfig
+func GetCloudTargetEnv(location string) string {
```

Comment:
there is a potential bug here. "The rest of the locations" are not public azure. there are USGov, Germany, etc... :(